dev-docs/source/Architecture.rst

```diff
@@ -0,0 +1,186 @@
+   :alt: Component diagram of mantidqt container
+
+   Container diagram for mantidqt Python package
+
+``mantidqt`` is a Python package defined to house a collection of Qt widgets
+that sit on top of ``mantid`` functionality.
+They provide a reusable set of widgets that can be used across multiple
+interfaces to give a consistent user experience when presenting given
+``mantid`` functionality to users. One example of such a widget is the
+``mantidqt.widgets.filefinder``, which sits on top of the framework
+``FileFinder``, and presents GUI elements to pass through and run the
+the framework logic.
+Providing a reusable widget gives a consistent user experience across
+multiple interfaces/applications when displaying ``mantid`` features to users.
+
+The package contains basic widgets, such as the aforementioned ``filefinder``,
+but also more advanced widgets such as
+:ref:`slice viewer <sliceviewer>` and
+the OpenGL-based :ref:`instrument viewer <InstrumentViewer>`,
+which combine many features of the framework together to provide advanced
+visualization to users.
+
+The package contains a mixture of C++ and Python code.
+It is defined in the ``qt/python/mantidqt`` directory of the source tree.
+Widgets written in C++ use the `sip <sip-ref_>`_ system to export the
+C++ classes to a Python module.
+The raw exported modules are all prefixed with an underscore,
+e.g. ``_common``,
+and classes are pulled into a pure-python module to decouple
+the C++ implementations from users.
+This allows for flexibility with the layout of the C++ classes/libraries
+without impacting users in the future.
+Clients of ``mantidqt`` should never import an underscored module directly.
+
+workbench
+#########
+
+.. figure:: /images/architecture-workbench.png
+   :scale: 85%
+   :align: center
+   :alt: Component diagram of workbench container
+
+   Container diagram for workbench application
+
+``workbench`` is an application written in pure Python in the ``qt/applications/workbench``
+directory of the source.
+It is not intended to be a library imported by others and as such offers
+no guarantees of API stability.
+It makes heavy use of the ``mantidqt`` widgets collection and puts them
+together to build the application shipped to users as ``MantidWorkbench``.
+
+The application provides some widgets that are only applicable to itself,
+such as the about screen, settings screen and project recovery mechanism that
+is specific to workbench.
+
+Plotting within ``workbench`` uses matplotlib_.
+The application bundle provides some custom features to step in on the backend,
+in ``workbench.plotting``, so that figures displayed within workbench offer
+a richer experience than a standard matplotlib figure,
+e.g. interactive fitting to data within a figure.
+The aim of these customizations is to be transparent to a user in terms of the
+matplotlib api.
+A plotting script should function, without modification,
+outside of workbench with the standard matplotlib backends,
```

Comment:
```suggestion
outside of workbench with the standard matplotlib backends, including jupyter notebooks
```
Perhaps we could also make explicit in the Framework/Python API section, that mantid as a framework can be used in vanilla python in batch mode or in any interactive python environment, including jupyter notebooks.

Reply:
Yes that's true. Being more explicit is worth it here.